src/rpcmining.cpp

```diff
@@ -85,14 +85,18 @@ UniValue getmininginfo(const UniValue& params, bool fHelp)
     obj.pushKV("MyNeuralHash", NN::GetNeuralHash());
 
     obj.pushKV("CPID",msPrimaryCPID);
-    obj.pushKV("RSAWeight",GetRSAWeightByCPID(msPrimaryCPID));
 
+    if (IsResearcher(msPrimaryCPID))
     {
-        double dMagnitudeUnit = GRCMagnitudeUnit(nTime);
-        double dAccrualAge,AvgMagnitude;
-        int64_t nBoinc = ComputeResearchAccrual(nTime, msPrimaryCPID, "getmininginfo", pindexBest, false, 69, dAccrualAge, dMagnitudeUnit, AvgMagnitude);
-        obj.pushKV("Magnitude Unit",dMagnitudeUnit);
-        obj.pushKV("BoincRewardPending",nBoinc/(double)COIN);
+        obj.pushKV("RSAWeight",GetRSAWeightByCPID(msPrimaryCPID));
```

Comment:
Why RSAWeight is displayed even for INVESTOR? Move it down to the conditional.
Or remove it, it is deprecated after all.